File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        int n = deck.length; 
        Queue<Integer> q = new LinkedList<>();
        
        for(int i=0; i<n;i++){
            q.add(i);
        }
        
        Arrays.sort(deck);
        
        int[] res = new int[n];
        for(int i=0; i<n; i++){
            res[q.poll()] = deck[i];
            q.add(q.poll()); 
        }
        
        return(res);
    }
}

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        n = len(deck)
        q = deque()
        
        for i in range(n):
            q.append(i)
        
        deck.sort()
        
        res = [0]*n
        
        for card in deck: 
            res[q.popleft()] = card
            
            if q:
                q.append(q.popleft())
                
        return res